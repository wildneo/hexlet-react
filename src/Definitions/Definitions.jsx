/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import uniqueId from 'lodash.uniqueid';

export default class Definitions extends React.Component {
  render() {
    const { data } = this.props;
    const f = (item) => (
      <React.Fragment key={uniqueId()}>
        <dt key={uniqueId()}>{item.dt}</dt>
        <dd key={uniqueId()}>{item.dd}</dd>
      </React.Fragment>
    );

    return (
      data.length === 0 ? null : <dl>{data.map(f)}</dl>
    );
  }
}
