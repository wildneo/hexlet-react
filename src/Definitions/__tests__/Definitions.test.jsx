import React from 'react';
import renderer from 'react-test-renderer';
import Definitions from '../Definitions';

test('Definitions 1', () => {
  const definitions = [
    { dd: 'one', dt: 'two' },
    { dd: 'another term', dt: 'another description' },
  ];
  const component = renderer.create(<Definitions data={definitions} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Definitions 2', () => {
  const definitions = [
    { dd: 'one', dt: 'two' },
  ];
  const component = renderer.create(<Definitions data={definitions} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Definitions 3', () => {
  const definitions = [
    { dd: 'another term', dt: 'another description' },
    { dd: 'one', dt: 'two' },
  ];
  const component = renderer.create(<Definitions data={definitions} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Definitions 4', () => {
  const definitions = [];
  const component = renderer.create(<Definitions data={definitions} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
