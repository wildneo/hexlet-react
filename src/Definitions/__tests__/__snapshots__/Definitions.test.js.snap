// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Definitions 1 1`] = `
<dl>
  <dt>
    two
  </dt>
  <dd>
    one
  </dd>
  <dt>
    another description
  </dt>
  <dd>
    another term
  </dd>
</dl>
`;

exports[`Definitions 2 1`] = `
<dl>
  <dt>
    two
  </dt>
  <dd>
    one
  </dd>
</dl>
`;

exports[`Definitions 3 1`] = `
<dl>
  <dt>
    another description
  </dt>
  <dd>
    another term
  </dd>
  <dt>
    two
  </dt>
  <dd>
    one
  </dd>
</dl>
`;

exports[`Definitions 4 1`] = `null`;
